src/optimizer/query_rewrite.c

```diff
@@ -1844,6 +1851,63 @@ qo_reduce_equality_terms (PARSER_CONTEXT * parser, PT_NODE * node, PT_NODE ** wh
       *orgp = parser_append_node (join_term_list, *orgp);
     }
 
+  /* remove always-true term */
+  while ((expr = ((expr_prev) ? expr_prev->next : *orgp)))
+    {
+      PT_OP_TYPE op = expr->info.expr.op;
+      cut_off = false;
+      opd1 = expr->info.expr.arg1;
+      opd2 = expr->info.expr.arg2;
+
+      if (expr->or_next == NULL)
+	{
+	  if (opd1 && opd2 && op == PT_EQ && opd1->node_type == PT_VALUE && opd2->node_type == PT_VALUE)
+	    {
+	      domain = pt_xasl_node_to_domain (parser, expr);
+	      qualifier = expr->info.expr.qualifier;
+	      dbv1 = pt_value_to_db (parser, opd1);
+	      dbv2 = pt_value_to_db (parser, opd2);
+	      opd3 = expr->info.expr.arg3;
+	      dbv3 = pt_value_to_db (parser, opd3);
```

Comment:
PT_EQ does not use arg3.
So, I think it is okay to pass dbv3 and opd3 as NULL here.
```
pt_evaluate_db_value_expr (parser, expr, expr->info.expr.op, dbv1, dbv2, NULL, &dbval_res, domain, opd1, opd2, NULL, qualifier)
```

If necessary for the future, how about changing it to the following?
- We have a problem calling the pt_value_to_db function without checking `opd3->node_type == PT_VALUE`.
```suggestion
	      dbv3 = (opd3 != NULL && opd3->node_type == PT_VALUE) ? pt_value_to_db (parser, opd3) : NULL;
```